eblocker-icapserver/src/test/java/org/eblocker/server/app/EblockerServerAppTest.java

```diff
@@ -0,0 +1,157 @@
+
+            try {
+                //When
+                sut.bootPhase(module);
+
+                //Then
+            } catch (Throwable e) {
+                fail();
+            }
+        }
+
+        @Test
+        void bootPhase_OnThrowable_late() {
+            //Given
+            doAnswer(invocationOnMock -> {
+                invocationOnMock.callRealMethod();
+                throw new RuntimeException();
+            }).when(sut).startHttpServer();
+            when(systemStatusService.starting(any())).thenReturn(systemStatusService);
+
+            //When
+            sut.bootPhase(module);
+
+            //Then
+            verify(systemStatusService).setExecutionState(ExecutionState.ERROR);
+            verify(systemStatusService).error(SubSystem.EBLOCKER_CORE);
+            verify(startupStatusReporter).startupFailed(any());
+
+        }
+    }
+
+    @Nested
+    class runtimePhase {
+
+        @BeforeEach
+        void setUp() {
+            when(systemStatusService.starting(any())).thenReturn(systemStatusService);
+            sut.bootPhase(module);
+        }
+
+        @Test
+        void runtimePhase_OnException_early() {
+            //Given
+            doThrow(RuntimeException.class).when(sut).initRuntimePhase();
+
+            //When
+            sut.runtimePhase();
+
+            //Then
+            verify(systemStatusService).setExecutionState(ExecutionState.ERROR);
+            verify(systemStatusService).error(SubSystem.EBLOCKER_CORE);
+            verify(startupStatusReporter).startupFailed(any());
+        }
+
+        @Test
+        void runtimePhase_OnException_late() {
+            //Given
+            doReturn("anyVersion").when(sut).doCheckSchemaVersion();
+
+            doAnswer(invocationOnMock -> {
+                invocationOnMock.callRealMethod();
+                throw new RuntimeException();
+            }).when(sut).startupCompleted();
+
+            //When
+            sut.runtimePhase();
+
+            //Then
+            verify(systemStatusService).setExecutionState(ExecutionState.ERROR);
+            verify(systemStatusService).error(SubSystem.EBLOCKER_CORE);
+            verify(startupStatusReporter).startupFailed(any());
+        }
+    }
+
+    private class EblockerModuleMock extends AbstractModule {
+
+        public EblockerModuleMock() {
+        }
+
+        @Override
+        protected void configure() {
+            super.configure();
+            bind(SystemStatusService.class).toInstance(systemStatusService);
+            bind(StartupStatusReporter.class).toInstance(startupStatusReporter);
+            bind(SubSystemServiceIndex.class).toInstance(mock(SubSystemServiceIndex.class));
+            bind(EventLogger.class).toInstance(mock(EventLogger.class));
+            bind(ShutdownService.class).toInstance(mock(ShutdownService.class));
+            bind(String.class).annotatedWith(Names.named("project.version")).toInstance("anyVersion");
```

Comment:
The following lines are required for the test to run on github. Without these, it will not find the configuration files that we naturally have locally